mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java

```diff
@@ -197,42 +198,66 @@ public MessageUid findFirstUnseenMessageUid(Mailbox mailbox) throws MailboxExcep
     }
 
     @Override
-    public Map<MessageUid, MessageMetaData> expungeMarkedForDeletionInMailbox(Mailbox mailbox, MessageRange set)
-            throws MailboxException {
+    public List<MessageUid> retrieveMessagesMarkedForDeletion(Mailbox mailbox, MessageRange messageRange) throws MailboxException {
         try {
-            final Map<MessageUid, MessageMetaData> data;
-            final List<MailboxMessage> results;
-            final MessageUid from = set.getUidFrom();
-            final MessageUid to = set.getUidTo();
             JPAId mailboxId = (JPAId) mailbox.getMailboxId();
+            List<MailboxMessage> messages = findDeletedMessages(messageRange, mailboxId);
+            return getUidList(messages);
+        } catch (PersistenceException e) {
+            throw new MailboxException("Search of MessageRange " + messageRange + " failed in mailbox " + mailbox, e);
+        }
+    }
+
+    @Override
+    public Map<MessageUid, MessageMetaData> deleteMessages(Mailbox mailbox, List<MessageUid> uids) throws MailboxException {
+        JPAId mailboxId = (JPAId) mailbox.getMailboxId();
+        Map<MessageUid, MessageMetaData> data = new HashMap<>();
+        List<MessageRange> ranges = MessageRange.toRanges(uids);
+
+        ranges.forEach(range -> {
+            List<MailboxMessage> messages = findDeletedMessages(range, mailboxId);
+            data.putAll(createMetaData(messages));
+            deleteDeletedMessages(range, mailboxId);
+        });
+
+        return data;
+    }
+
+    private List<MailboxMessage> findDeletedMessages(MessageRange messageRange, JPAId mailboxId) {
+        MessageUid from = messageRange.getUidFrom();
+        MessageUid to = messageRange.getUidTo();
+
+        switch (messageRange.getType()) {
+            case ONE:
+                return findDeletedMessagesInMailboxWithUID(mailboxId, from);
+            case RANGE:
+                return findDeletedMessagesInMailboxBetweenUIDs(mailboxId, from, to);
+            case FROM:
+                return findDeletedMessagesInMailboxAfterUID(mailboxId, from);
+            case ALL:
+            default:
```

Comment:
If the type of range is onknown it is safer to throw a runtime exception so that the programmer get that something is wrong rather than trying to enforce wrong behaviours....